fdb-extensions/src/test/java/com/apple/foundationdb/async/MoreAsyncUtilTest.java

```diff
@@ -133,7 +138,12 @@ public void getWithDeadlineRunsOnExecutor() throws ExecutionException, Interrupt
                         return Thread.currentThread().getName();
                     })
                     .get();
-            assertEquals("test-deadline-exceeded-thread-0", callbackThreadName);
+
+            // Most of the time, the callback should come fom the scheduledExecutor. However, if there's some hiccup and actually
+            // setting up the future chain takes longer than the deadline time, then it's possible for the callback to complete on
+            // the test worker thread
+            assertThat("Callback should have been executed on thread managed by scheduled executor or by calling thread",
```

Comment:
This reduces the thoroughness of this test.
Basically, if it reliably took 5ms to setup the future chain (I've seen two test runs fail this way), it would no longer know whether it actually used the provided `scheduledExecutor`.
Would it make sense to increase the deadline to like 15ms.
Would it help to have the `scheduledExecutor` run something before this to make sure the thread pool is fully primed?

Reply:
Yeah, I'd thought about how it reduces the thoroughness. I wasn't super concerned about it, because locally, it seems to pretty reliably land on the scheduled executor, so it's not like that isn't getting coverage. I did think about just increasing the deadline, but I didn't know what a reasonable value would be for that. (I thought 5 ms was going to be enough, but alas, that was incorrect.) Part of that is I'm not actually sure what it's waiting on. I don't _think_ it's actually the `scheduledExecutor` in this case, because that future is the one that's completing (too) quickly, and it's some part of `CompletableFuture.anyOf`.

I guess we could make this a repeated test, as it seems unlikely that running this test, say, 10 times, none of them would wind up executing on the scheduled executor.

The main thing I was worried about was that somehow, the `getWithDeadline` would wind up executing its callback on some other scheduled executor pool (like the default one), and this test does give us some coverage of that.